lib/engine/game/g_1846/game.rb

```diff
@@ -631,6 +641,30 @@ def remove_lm_icons
           end
         end
 
+        def remove_boomtown_icons
+          self.class::BOOMTOWN_HEXES.each do |hex|
+            hex_by_id(hex).tile.icons.reject! { |icon| icon.name == self.class::BOOMTOWN_ICON }
+          end
+        end
```

Comment:
These `remove_*_icons` method are a little repetitive, could they be changed to something like this?

```suggestion
        def remove_icons(hex_ids, icon_name)
          hex_ids.each do |hex_id|
            hex_by_id(hex_id).tile.icons.reject! { |icon| icon.name == icon_name}
          end
        end

        def remove_boomtown_icons
          remove_icons(hex_ids,  self.class::BOOMTOWN_ICON)
        end
```

Reply:
Sure, I agree w/ the DRY violation there

Do you know a ruby-ish way to obtain `hex_ids` directly from the company instead of having some duplicative vars / hash?
e.g. `steamboat.ability['assign_hexes'].hexes`